week2/react-exercises/src/components/DogGallery.js

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from 'react'
+import Button from './DogButton'
+import DogPhoto from './DogPhoto'
+
+function DogGallery() {
+    const [dogPhotos, setDogPhotos] = useState([])
+    const [loading, setLoading] = useState(false)
+    const [image, setImage] = useState(false)
+
+    async function getDogPhoto() {
+        const response = await fetch('https://dog.ceo/api/breeds/image/random')
+        if (response.ok) {
+            const data = await response.json()
+            const url = [...dogPhotos, data]
+            setDogPhotos(url)
+            setLoading(true)
+            setImage(true)
+        } else {
+            throw new Error("Photo loading failed.")
+        }
+    }
+
+    return (
+        <div className="dogGallery">
+            {image === false && <h2 className="notes"> Get your first dog by clicking the button! </h2>}
+            <Button onClick={() => getDogPhoto()} />
+            <div className="imgList">
+                {loading === true && dogPhotos.map(img => <DogPhoto url={img.message} />)}
```

Comment:
This element is part of an array since it's a `.map` return, and as an addition keys are like `id` attributes in HTML but that how it works with React, however as a reminder from the past review using keys is only in the top level element would get it's benefit, and keys are not accessible from inside the component.
```suggestion
                {loading === true && dogPhotos.map((img, index) => <DogPhoto url={img.message} key={index}  />)}
```

using index fine for now because we don't have an id in the returned value, however using some else thing as `uuid` would be preferred.